sdk/python/resources/connections/connections.ipynb

```diff
@@ -0,0 +1,217 @@
+    "\n",
+    "To connect to a workspace, we need identifier parameters - a subscription, resource group and workspace name. We will use these details in the `MLClient` from `azure.ai.ml` to get a handle to the required Azure Machine Learning workspace. We use the default [default azure authentication](https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) for this tutorial. Check the [configuration notebook](../../jobs/configuration.ipynb) for more details on how to configure credentials and connect to a workspace."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Enter details of your AML workspace\n",
+    "subscription_id = \"<SUBSCRIPTION_ID>\"\n",
+    "resource_group = \"<RESOURCE_GROUP>\"\n",
+    "workspace = \"<AML_WORKSPACE_NAME>\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get a handle to the workspace\n",
+    "from azure.ai.ml import MLClient\n",
+    "from azure.identity import DefaultAzureCredential\n",
+    "\n",
+    "ml_client = MLClient(\n",
+    "    DefaultAzureCredential(), subscription_id, resource_group, workspace\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2 Create Workspace Connection\n",
+    "An Azure Machine Learning workspace is secure and trustworthy way for AML users to store credentials to access external resources from AML\n",
+    "\n",
+    "Workspace Connection can be created using the `WorkspaceConnection` class. This class accepts the following key parameters:\n",
+    "- `name` - Name of connection.\n",
+    "- `target` - The URL or ARM resource ID of the external resource.\n",
+    "- `credentials` - The credentials for authenticating to the external resource.\n",
+    "- `type` - The category of external resource for this connection.\n",
+    "\n",
+    "## 2.1 Create a Workspace Connection\n",
+    "In this example we will create a **Workspace Connection** to an [**Azure Container Registry**]([ACR](https://azure.microsoft.com/en-us/products/container-registry/)) and use **username/password** credentials\n",
+    "\n",
+    "Once connection is successfully created you can reference images from [Azure Container Registry]([ACR](https://azure.microsoft.com/en-us/products/container-registry/)) in AML environment and AML will automatically fetch these credentials while building the environment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import WorkspaceConnection\n",
+    "from azure.ai.ml.entities import UsernamePasswordConfiguration\n",
+    "\n",
+    "credentials = UsernamePasswordConfiguration(\n",
+    "    username=\"<user_name_for_acr>\",\n",
+    "    password=\"<pwd_for_acr>\"\n",
```

Comment:
We should educate users not to hardcode passwords, but to use environment variables instead.